6.0/stretch/Dockerfile

```diff
@@ -33,79 +25,108 @@ ENV VMOD_BUILD_DEPS \
 
 # persistent / runtime deps
 RUN apt-get update && apt-get install -y \
-		gcc \
-		libc6-dev \
+		apt-transport-https \
+		ca-certificates \
 	--no-install-recommends && rm -r /var/lib/apt/lists/*
 
-ENV VARNISH_VERSION 6.0.3
-ENV VARNISH_URL https://varnish-cache.org/_downloads/varnish-6.0.3.tgz
-ENV VARNISH_SHA256 4e0a4803b54726630719a22e79a2c5b36876506497e24fb39a47e9df219778d7
-
 RUN set -eux; \
 	\
 	fetchDeps=' \
-		ca-certificates \
 		wget \
 	'; \
-	buildDeps=" \
-		$VMOD_BUILD_DEPS \
-		dpkg-dev \
-		libedit-dev \
-		libjemalloc-dev \
-		libncurses5-dev \
-		libpcre3-dev \
-	"; \
-	savedAptMark="$(apt-mark showmanual)"; \
+	if ! command -v gpg > /dev/null; then \
+		fetchDeps="$fetchDeps \
+			dirmngr \
+			gnupg \
+		"; \
+	fi; \
 	apt-get update; \
-	apt-get install -y --no-install-recommends $fetchDeps $buildDeps; \
+	apt-get install -y --no-install-recommends $fetchDeps; \
 	rm -rf /var/lib/apt/lists/*; \
 	\
-	wget -O varnish.tar.gz "$VARNISH_URL"; \
+	export GNUPGHOME="$(mktemp -d)"; \
+# key is not found on any keyserver, but we match its fingerprint
+	wget -O - 'https://packagecloud.io/varnishcache/varnish60lts/gpgkey' | gpg --batch --import; \
+	gpg --batch --export 'DD2C378724BD39C18AAA47FE3AEAFFBB82FBBA5F' > /etc/apt/trusted.gpg.d/varnish.gpg; \
+	command -v gpgconf > /dev/null && gpgconf --kill all; \
+	rm -rf "$GNUPGHOME"; \
+	apt-key list; \
 	\
-	if [ -n "$VARNISH_SHA256" ]; then \
-		echo "$VARNISH_SHA256 *varnish.tar.gz" | sha256sum -c -; \
-	fi; \
+	apt-get purge -y --auto-remove -o APT::AutoRemove::RecommendsImportant=false $fetchDeps
+
+ENV VARNISH_PACKAGE_VERSION 6.0.2-1~stretch
+
+RUN set -eux; \
 	\
-	mkdir -p /usr/src/varnish; \
-	tar -zxf varnish.tar.gz -C /usr/src/varnish --strip-components=1; \
-	rm varnish.tar.gz; \
+# see note below about "*.pyc" files
+	export PYTHONDONTWRITEBYTECODE=1; \
+	\
+	dpkgArch="$(dpkg --print-architecture)"; \
+	case "$dpkgArch" in \
+		amd64) \
+# arches officialy built by upstream
+			echo "deb https://packagecloud.io/varnishcache/varnish60lts/debian/ stretch main" > /etc/apt/sources.list.d/varnish.list; \
+			apt-get update; \
+			;; \
+		*) \
```

Comment:
is there a point in supporting non-amd64 arches?

Reply:
https://github.com/docker-library/official-images#multiple-architectures

Reply:
I think the official Docker images take the approach of blacklisting the architectures that wouldn't work: https://github.com/docker-library/php/blob/e63194a0006848edb13b7eff5a7f9d790d679428/generate-stackbrew-library.sh#L124-L129